From 7968ab66cc7c129511c16277e2089cd98f7c3945 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 1 Feb 2019 15:46:47 -0800
Subject: [PATCH 0508/1852] drm: xlnx: zynqmp: Handle the event when there's no
 fb

commit 0afa4e461d6d2e48529bac99abad490c927ac2bb from
https://github.com/Xilinx/linux-xlnx.git

The commit 846c7dfc1193 ("drm/atomic: Try to preserve the crtc enabled
state in drm_atomic_remove_fb, v2."), changes the behavior of event
handling by keeping the crtc as active when removing the fb. Previously
the crtc became inactive and the atomic helper completes the event
as the driver wouldn't. This requires the event to be consumed by
the driver, otherwise it gives warnings from the atomic helper seeing
the event. So consume the event when there's no fb attached to
the new state.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index e75348bb5331..8c74a227c09e 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -2942,7 +2942,7 @@ zynqmp_disp_crtc_atomic_begin(struct drm_crtc *crtc,
 	drm_crtc_vblank_on(crtc);
 	/* Don't rely on vblank when disabling crtc */
 	spin_lock_irq(&crtc->dev->event_lock);
-	if (crtc->primary->state->fb && crtc->state->event) {
+	if (crtc->state->event) {
 		/* Consume the flip_done event from atomic helper */
 		crtc->state->event->pipe = drm_crtc_index(crtc);
 		WARN_ON(drm_crtc_vblank_get(crtc) != 0);
-- 
2.31.1

